### PR TITLE
CI-CD Updates

### PR DESCRIPTION
**Pull request by Skptak**
Description
-----------
CI-CD Updates to use new versions of the CI-CD-Github-Actions
Individual PRs can be seen at:
[Complexity Check](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/70)
[Coverage-Cop](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/71)
[Executable Monitor](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/72)
[CSpell](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/75)
[Uncrustify Formatting Update](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/76)
[Link Verifier](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/81)

Test Steps
-----------

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,46 +126,61 @@ int main()
...
 ## Building unit tests
 
 ### Checkout Unity Submodule
-By default, the submodules in this repository are configured with `update=none` in [.gitmodules](.gitmodules), to avoid increasing clone time and disk space usage of other repositories (like [amazon-freertos](https://github.com/aws/amazon-freertos) that submodules this repository).
 
-To build unit tests, the submodule dependency of Unity is required. Use the following command to clone the submodule:
+By default, the submodules in this repository are configured with `update=none`
+in [.gitmodules](.gitmodules), to avoid increasing clone time and disk space
+usage of other repositories (like
+[amazon-freertos](https://github.com/aws/amazon-freertos) that submodules this
```
Comment (bradleysmith23) [inline README.md:159]: Isn't this repo deprecated? Should this still be mentioned?

File: cspell.config.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+---
+$schema: https://raw.githubusercontent.com/streetsidesoftware/cspell/main/cspell.schema.json
+version: '0.2'
+# Allows things like stringLength
+allowCompoundWords: true
```
Comment (bradleysmith23) [inline cspell.config.yaml:5]: Are both occurrences of allowCompoundWords necessary?

File: cspell.config.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+useGitignore: true
+languageSettings:
+  - allowCompoundWords: true
+    caseSensitive: false
+    enabled: true
+    languageId: c
+    locale: "*"
+# Could split this up? And do a dictionary for each repo?
```
Comment (bradleysmith23) [inline cspell.config.yaml:13]: Remove unnecessary comment.

**Review by bradleysmith23 — CHANGES_REQUESTED** [review]
Left a few comments, mostly repeats from of the comments left for FreeRTOS+TCP

File: README.md (hunk 1 above)
Comment (Skptak) [reply README.md:159]: Limiting this PR to just changes to the CI-CD actions, spelling, and formatting issues.
This link could be updated to mention FreeRTOS/FreeRTOS instead at this point, but that change is outside the scope of this PR.

File: cspell.config.yaml (hunk 2 above)
Comment (Skptak) [reply cspell.config.yaml:5]: Fixed.

File: cspell.config.yaml (hunk 3 above)
Comment (Skptak) [reply cspell.config.yaml]: Fixed.

**Review by archigup — APPROVED** [review]

**Review by bradleysmith23 — APPROVED** [review]

**Pull request merged**